master/internal/api_command.go

```diff
@@ -352,6 +361,7 @@ func (a *apiServer) LaunchCommand(
 		return nil, api.WrapWithFallbackCode(err, codes.InvalidArgument,
 			"failed to prepare launch params")
 	}
+	spec := launchReq.Spec
```

Comment:
I wrote the below comment before I read back up and saw `launchReq.Spec` was a pointer, but I'm gonna post it anyway as some hand wavy argument that this shallow is confusing / error-prone, anyway.

does this work? pretty sure this shallow copy is going to make some of the later mutates ineffective (e.g. `spec.Config.Description, see https://go.dev/play/p/sa4sIHVOELg)